akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/AccountExampleWithCommandHandlersInState.scala

```diff
@@ -145,4 +147,38 @@ object AccountExampleWithCommandHandlersInState {
   }
   //##account-entity
 
+  import akka.actor.testkit.typed.scaladsl._
+  class AccountExampleSpec extends ScalaTestWithActorTestKit() {
+    import AccountEntity._
+
+    val behavior = AccountEntity.behavior("1")
+    val ref = spawn(behavior)
+
+    proveTypeIs[Behavior[AccountCommand[Nothing]]](behavior)
+    proveTypeIs[ActorRef[AccountCommand[Nothing]]](ref)
+
+    val probe = TestProbe[OperationResult]
+    ref.tell(CreateAccount()(probe.ref))
+    probe.expectMessage(Confirmed)
+
+    val probe2 = TestProbe[CurrentBalance]
+    ref.tell(GetBalance()(probe2.ref))
+
+    {
+      val behavior = AccountEntity.behavior[OperationResult]("1")
```

Comment:
isn't it strange to define the reply type at use site like this? I see that `GetBalance` can't be sent below, which is nice but is there a real need for supporting that? The drawback I see is that if we would encourage that usage in documentation the signature of the factory method `AccountEntity.behavior` is more complicated than otherwise.

Reply:
Ideally, this would be the command itself and be possible by having Behavior be contravariant, but it's not.

I'll try to circle back to this and rebase these changes and tests on that PR.